src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -211,16 +211,32 @@ bool IPackageInstallerService.IsEnabled(ProjectId projectId)
             return true;
         }
 
-        protected override void EnableService()
+        private async ValueTask<IVsPackageSourceProvider> GetPackageSourceProviderAsync(CancellationToken cancellationToken)
+        {
+            Contract.ThrowIfFalse(IsEnabled);
+
+            if (!_packageSourceProvider.IsValueCreated)
+            {
+                // Switch to background thread for known assembly load
+                await TaskScheduler.Default;
+            }
+
+            return _packageSourceProvider.Value;
+        }
+
+        protected override async Task EnableServiceAsync(CancellationToken cancellationToken)
```

Comment:
Can the Platform team offer a ValueTask api? Is there a reason they don't do that for a lot of these?

Reply:
`ValueTask` wouldn't buy us much here. `async` methods that complete synchronously and return `Task` use a cached instance.